app/graphql/types/git/directory_entry_type.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength
+Types::Git::DirectoryEntryType = GraphQL::InterfaceType.define do
+  # rubocop:enable Metrics/BlockLength
+  name 'DirectoryEntry'
+  description 'A directory entry (directory or file) of a repository'
+
+  field :name, !types.String do
+    description 'The name of the entry'
+  end
+
+  field :path, !types.String do
+    description 'The path of the entry'
+  end
+
+  # rubocop:disable Metrics/BlockLength
+  field :log, !types[!Types::Git::CommitType] do
```

Comment:
Here is some official material on pagination: http://graphql.org/learn/pagination/